pkg/apply/resource_infos.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package apply
+
+import (
+	"sort"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+)
+
+type ResourceInfos []*resource.Info
+
+var _ sort.Interface = ResourceInfos{}
+
+func (a ResourceInfos) Len() int      { return len(a) }
+func (a ResourceInfos) Swap(i, j int) { a[i], a[j] = a[j], a[i] }
+func (a ResourceInfos) Less(i, j int) bool {
+	x := a[i].Object.GetObjectKind().GroupVersionKind()
+	o := a[j].Object.GetObjectKind().GroupVersionKind()
+	if !Equals(x, o) {
+		return IsLessThan(x, o)
+	}
+	return a[i].String() < a[j].String()
+}
+
+// An attempt to order things to help k8s, e.g.
+// a Service should come before things that refer to it.
+// Namespace should be first.
+// In some cases order just specified to provide determinism.
+var orderFirst = []string{
+	"Namespace",
+	"ResourceQuota",
+	"StorageClass",
+	"CustomResourceDefinition",
+	"MutatingWebhookConfiguration",
+	"ServiceAccount",
+	"PodSecurityPolicy",
+	"Role",
+	"ClusterRole",
+	"RoleBinding",
+	"ClusterRoleBinding",
+	"ConfigMap",
+	"Secret",
+	"Service",
+	"LimitRange",
+	"PriorityClass",
+	"Deployment",
+	"StatefulSet",
+	"CronJob",
+	"PodDisruptionBudget",
+}
+var orderLast = []string{
+	"ValidatingWebhookConfiguration",
+}
+var typeOrders = func() map[string]int {
```

Comment:
This might be more readable with a named function?

Reply:
Done